handbook.md

```diff
@@ -0,0 +1,233 @@
+
+### Contributors
+
+Write a good description, a good title, and explain why change is necessary. Just a line or two explaining the problem, the solution, and a way to reproduce the issue (extra points if it's a script) help enormously.
+
+Assume reviewers have no idea or background about your patch. Even if the usual reviewers know you personally and you know they know why your change is necessary, maybe not all reviewers are aware of it. Furthermore, after some time, they might not remember well, and new reviewers will not be able to review your code without background and a good explanation.
+
+Generally, if you want to submit significant changes to the code, discuss it first on #theforeman-dev (Freenode) or Foreman-dev (mailing-list). If you know who are the usual maintainers for the code you want to change, try to ask them to validate your assumptions, your design, and if they can, ask them to review your code. This will save you a lot of going back and forth with reviewers that do not understand the reasoning behind your pull request.
+
+Submit changes incrementally. If you are submitting a pull request that will break compatibility with older APIs, spend the effort to make it compatible first. If you think your feature is not ready for prime time yet, it's OK to submit small changes and hide the feature using feature flags.
+
+Feel free to bring the attention of reviewers by calling them using '@' on GitHub.
+
+### Reviewers
+
+#### Style
+
+Be open to change, but respect the borders between core and plugins. Refer to [What exactly is core?](handbook.html#Whatexactlyiscore?) for that.
+
+Be strict about tests. The project has rewuired tests for most changes to the Ruby code base, however we have not been historically as strict with UI changes. Ask for integration tests for any change that could break in future versions without tests. Testing is a powerful weapon to ensure regressions don't happen.
+
+Be mindful of the use of the word 'you'. It can be easily misunderstood as you reviewing the 'person' not the 'code'. Try not to personalize any references to code. I.e: 'you changed that in line 42' vs 'that change in line 42'.
+
+Back up your reviews by facts when they are not obvious or when contributors might not know why they should make some change. Adopt a "teaching" approach, but again, back up your 'teachings' with links to other sources.
+
+Use '@' to raise the attention of the contributor about any issue. If an user is not 'watching' the project on GitHub or is not subscribed to a particular issue, GitHub will not show notifications nor send emails to that user.
+
+If there is conflict, point to this handbook for reference.
+
+#### Checklist
+http://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches
+
+#### Labels
+
+Labels help reviewers understand what is the status of a pull request. Thanks to them, reviewers can make sure no pull request remains unreviewed and when they get updated, they see it.
```

Comment:
Also, unless GitHub fixed it and I missed it, setting/changing labels requires commit access, so new reviewers *may* not be able to change them.